core/src/main/java/org/apache/iceberg/SnapshotSummary.java

```diff
@@ -321,6 +323,23 @@ void merge(UpdateMetrics other) {
     }
   }
 
+  protected static Map<String, String> fetchTotalStatsFromSummary(Map<String, String> summary) {
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+    List<String> totalProps = Arrays.asList(
+            TOTAL_DATA_FILES_PROP,
+            TOTAL_DELETE_FILES_PROP,
+            TOTAL_EQ_DELETES_PROP,
+            TOTAL_POS_DELETES_PROP,
+            TOTAL_FILE_SIZE_PROP,
+            TOTAL_RECORDS_PROP);
```

Comment:
This creates a new list each time through the method. Instead, can you save this as a static final field?

Reply:
agreed, removing this entire method due to above comment changes.